hystrix/commandbuilder/command_builder.go

```diff
@@ -0,0 +1,102 @@
+package commandbuilder
+
+import (
+	"time"
+
+	"github.com/myteksi/hystrix-go/hystrix"
+)
+
+// CommandBuilder builder for constructing new command
+
+type CommandBuilder struct {
+	commandName            string
+	timeout                int
+	commandGroup           string
+	maxConcurrentRequests  int
+	requestVolumeThreshold int
+	sleepWindow            int
+	errorPercentThreshold  int
+	// for more details refer - https://github.com/Netflix/Hystrix/wiki/Configuration#maxqueuesize
+	queueSizeRejectionThreshold *int
+}
+
+// New Create new command
+func New(commandName string) *CommandBuilder {
+	return &CommandBuilder{
+		commandName:            commandName,
+		timeout:                hystrix.DefaultTimeout,
+		commandGroup:           commandName,
+		maxConcurrentRequests:  hystrix.DefaultMaxConcurrent,
+		requestVolumeThreshold: hystrix.DefaultVolumeThreshold,
+		sleepWindow:            hystrix.DefaultSleepWindow,
+		errorPercentThreshold:  hystrix.DefaultErrorPercentThreshold,
+	}
+}
+
+// WithTimeout modify timeout
+func (cb *CommandBuilder) WithTimeout(timeoutInMs int) *CommandBuilder {
+	cb.timeout = timeoutInMs
+	return cb
+}
+
+// WithCommandGroup modify commandGroup
+func (cb *CommandBuilder) WithCommandGroup(commandGroup string) *CommandBuilder {
+	cb.commandGroup = commandGroup
+	return cb
+}
+
+// WithMaxConcurrentRequests modify max concurrent requests
+// if not already set, this will also set the queue size as 5 times the max concurrent requests
+func (cb *CommandBuilder) WithMaxConcurrentRequests(maxConcurrentRequests int) *CommandBuilder {
+	cb.maxConcurrentRequests = maxConcurrentRequests
+	if cb.queueSizeRejectionThreshold == nil {
+		queueSize := 5 * cb.maxConcurrentRequests
```

Comment:
Why there is still a hard coded parameter? I thought this could be using the attribute of the circuit directly? Also, how to handle circuits that are opt-out of this feature?

Reply:
I believe the queue size should be independent of max concurrency. Or if they are related, we can derive that directly instead of having an additional  attribute to record it.

Reply:
If a client wants to opt-out of the feature, they can specify `WithQueueSize(0)` (see unit test) or if they want a different value for the buffer `WithQueueSize(1024)`.

IMO, QueueSize is a function of maxConcurrency (probably timeout and average response time as well). I don't know of a better default value for the queueSize. The additional attribute helps in providing a better value by the clients, if the default does not work for them.